### PR TITLE
Feature/mute and device selection

### PR DESCRIPTION
**Pull request by hasevr**
Now mute/unmute and device selection of mic/speaker/camera work well.
This request comes to a bit large but most part is already reviewed and fixed.

### CODE REVIEW

File: services/conference/src/scripts/components/footer/footer.tsx [hunk 1]
```diff
@@ -30,53 +32,135 @@ const useStyles = makeStyles((theme) => {
...
+      color: 'primary',
+    },
+    item:{
+    },
   })
 })
 
 
 export const Footer: React.FC<BaseProps> = (props) => {
   const classes = useStyles()
```
Comment (zlguo1996) [inline services/conference/src/scripts/components/footer/footer.tsx:45]: Component 'Footer' seems be able to split into three child component: MicButton, SpeakerButton, VideoButton. Every button contains two fabs and one menu.

File: services/conference/src/scripts/models/TrackMiddleware/ChooseDevice.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import participants from '@stores/participants/Participants'
+import JitsiMeetJS, {JitsiLocalTrack, JitsiTrackOptions, JitsiValues} from 'lib-jitsi-meet'
+import {reaction} from 'mobx'
+
+function replaceTrack(newTrack:JitsiLocalTrack) {
+  const oldTracks = connection.conference?.getLocalTracks(newTrack.getType())
+  if (oldTracks !== undefined) {
+    connection.conference?.replaceTrack(oldTracks[0], newTrack)
+  }
+  const did_ = newTrack.getTrack().getSettings().deviceId
```
Comment (zlguo1996) [inline services/conference/src/scripts/models/TrackMiddleware/ChooseDevice.ts:12]: 'deviceId'

File: services/conference/src/scripts/models/TrackMiddleware/ChooseDevice.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  if (newTrack.getType() === 'audio') {
+    participants.local.get().devicePreference.audioInputDevice = did
+  }else if (newTrack.getType() === 'video') {
+    participants.local.get().devicePreference.videoInputDevice = did
+  }
+}
+
+reaction(
+  () => participants.local.get().devicePreference.audioInputDevice,
+  (did) => {
```
Comment (zlguo1996) [inline services/conference/src/scripts/models/TrackMiddleware/ChooseDevice.ts:23]: 'deviceId'

File: services/conference/src/scripts/models/TrackMiddleware/ChooseDevice.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    JitsiMeetJS.createLocalTracks({devices:['audio'], micDeviceId: did}).then(
+      (tracks: JitsiLocalTrack[]) => { replaceTrack(tracks[0]) },
+    )
+  },
+)
+
+reaction(
+  () => participants.local.get().devicePreference.videoInputDevice,
+  (did) => {
```
Comment (zlguo1996) [inline services/conference/src/scripts/models/TrackMiddleware/ChooseDevice.ts:32]: 'deviceId'

File: services/conference/src/scripts/models/TrackMiddleware/ChooseDevice.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        replaceTrack(tracks[0])
+      },
+    )
+  },
+)
+
+reaction(
+  () => participants.local.get().devicePreference.audioOutputDevice,
+  (deviceId) => {
+    JitsiMeetJS.mediaDevices.setAudioOutputDevice(deviceId)  //  this not works
```
Comment (zlguo1996) [inline services/conference/src/scripts/models/TrackMiddleware/ChooseDevice.ts:44]: If not works, we can delete the whole reaction

File: .vscode/settings.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "editor.tabSize": 2,
-  "editor.detectIndentation": false,
+  "editor.detectIndentation": true,
```
[the remark this answers is not in the document]
Comment (hasevr) [reply .vscode/settings.json:3]: On my environment, sometimes, same file load through different path and cause problem when save.
I just expect this option solve this problem.

File: services/conference/src/scripts/components/footer/footer.tsx [hunk 7]
```diff
@@ -30,53 +32,135 @@ const useStyles = makeStyles((theme) => {
...
+  const participants = useParticipantsStore()
 
   const [micMenuEl, setMicMenuEl] = React.useState<Element|null>(null)
-  const closeMicMenu = () => { setMicMenuEl(null) }
+  const closeMicMenu = (did:string) => {
+    if (did) { participants.local.get().devicePreference.audioInputDevice = did }
+    setMicMenuEl(null)
+  }
+  const [speakerMenuEl, setSpeakerMenuEl] = React.useState<Element|null>(null)
+  const closeSpeakerMenu = (did:string) => {
```
[the remark this answers is not in the document]
Comment (hasevr) [reply services/conference/src/scripts/components/footer/footer.tsx:54]: this also very short life variable and I don't think it needs long name.

**Pull request merged**